### DISCUSSION TITLE
Comma as a decimal separator

### DISCUSSION THREAD

**Discussion by J-Duelund** [KaTeX/KaTeX#3744] [Q&A] [answered]
Hi!

Is it possible to have a comma as a decimal separator, meaning that there isn't added extra space after the comma? When using LaTeX I would use the icomma package.

I know that I can write 3{,}14159... etc., but when writing a lot of decimal numbers, it would be easier to make an overall change if possible.

**Comment by edemaine** [comment, marked as answer]
You could use ``\def,{{\char`,}}`` or the setting ``{macros: {",": "{\\char`,}"}}`` to permanently redefine `,` to `{,}`. But this makes it hard to form lists.

[icomma.sty's behavior](https://mirror.las.iastate.edu/tex-archive/macros/latex/contrib/was/icomma.pdf) is to add spacing when the comma is followed by a space.  I believe this is implemented by the following setting:

```js
{macros: {",": (context) => context.future()?.text === " " ? "{\\char`,}" : "\\char`,"}
```

I don't think this can currently be expressed via `\def`, but hopefully this suffices for your purposes.

**Reply by J-Duelund** [reply]
Thanks for the answer. I think that it will work.
